### PR TITLE
Add multi flow tests for KB, kamelets and integrations 

### PR DESCRIPTION
**Pull request by tplevko**
Found the flow parsing was adding the '---' at the end of parsed yaml I think it might be cut off.

@Delawen please have a look.

### CODE REVIEW

**Review by Delawen — APPROVED** [review]

**Comment by Delawen** [comment]
Good catch!

**Comment by Delawen** [comment]
Could be some issue with new lines in windows? Tests fail too much.

File: camel-route-support/src/test/java/io/kaoto/backend/api/service/deployment/generator/camelroute/DeploymentGeneratorServiceTest.java [hunk 1]
```diff
@@ -48,6 +48,43 @@ void parse() {
...
+                + "        period: '5000'\n"
+                + "      steps:\n"
+                + "      - to:\n"
+                + "          uri: log:tick\n";
+
+        var parsed = stepParserService.getParsedFlows(yaml);
+
+        var yaml2 = deploymentGeneratorService.parse(parsed);
+
+        assertEquals(yaml, yaml2);
```
Comment (Delawen) [inline camel-route-support/src/test/java/io/kaoto/backend/api/service/deployment/generator/camelroute/DeploymentGeneratorServiceTest.java:85]: Can you use `assertThat(yaml).isEqualToNormalizingNewlines(yaml2)` instead?

**Review by Delawen — CHANGES_REQUESTED** [review]

**Comment by tplevko** [comment]
@Delawen I updated also the KameletBinding and Kamelets DeploymentGeneratorServices.

I'm still trying to bypass the windows issues, the `isEqualToNormalizingNewlines` doesn't seem to have helped, so trying other things too

**Comment by tplevko** [comment]
This should be ready for review - the `native tests for windows-latest` failure seems to be unrelated to this PR

**Review by Delawen — APPROVED** [review]

**Pull request merged**